setup_chef_server.sh

```diff
@@ -20,51 +20,30 @@ if [[ -z "$CURL" ]]; then
   exit
 fi
 
-if [[ ! -f /etc/apt/sources.list.d/bcpc.list ]]; then
-    # Create an Apt repo entry
-    echo "deb [arch=amd64] file://$(pwd)/bins/ 0.5.0 main" > \
-	 /etc/apt/sources.list.d/bcpc.list
-    # add repo key
-    apt-key add bins/apt_key.pub
-    # update only the BCPC local repo
-    apt-get -o Dir::Etc::SourceList=/etc/apt/sources.list.d/bcpc.list \
-	    -o Dir::Etc::SourceParts="-" \
-	    -o APT::Get::List-Cleanup="0" \
-	    update
-fi
-
-# It's important to install the chefdk before chef, so that
-# /usr/bin/knife and /usr/bin/chef-client are symlinks into /opt/chef
-# instead of /opt/chefdk.
-apt-get -y install chefdk
-apt-get -y install chef=11.12.8-2
-
-GEM_PATH=/opt/chef/embedded/bin/gem
-
-if [ ! -x $GEM_PATH ]; then
-   echo "$GEM_PATH is not executable!"
-   echo "Cannot configure chef gem source!"
-   exit -1
-fi
+#
+# build_bins.sh will run chef in local mode in order to create the
+# BCPC local repo.
+#
+./build_bins.sh
 
-if ! $GEM_PATH sources | grep $(pwd)/bins; then
-   $GEM_PATH sources --add file://$(pwd)/bins/
-fi
+# Make the bootstrap a client of its own BCPC local repo
+echo "deb [trusted=yes arch=amd64] file://$(pwd)/bins/ 0.5.0 main" > \
+     /etc/apt/sources.list.d/bcpc.list
 
-if $GEM_PATH sources | grep rubygems.org; then
-   $GEM_PATH sources --remove http://rubygems.org/
-fi
+# Update only the BCPC local repo
+apt-get -o Dir::Etc::SourceList=/etc/apt/sources.list.d/bcpc.list \
+	-o Dir::Etc::SourceParts="-" \
+	-o APT::Get::List-Cleanup="0" \
+	update
 
-# setup OpenStack hint
-mkdir -p /etc/chef/ohai/hints/
-touch /etc/chef/ohai/hints/openstack.json
+apt-get -y install chef=11.18.12-1
 
 if dpkg -s chef-server 2>/dev/null | grep -q ^Status.*installed && \
    dpkg -s chef 2>/dev/null | grep -q ^Version.*11; then
   echo chef-server is installed
 else
   apt-get -y install chef-server
-  mkdir /etc/chef-server
+  mkdir -p /etc/chef-server
```

Comment:
Exactly the opposite.  If `/etc/chef-server` is already present, a `mkdir` without `-p` will return an error.